src/mlpack/tests/main_tests/linear_regression_test.cpp

```diff
@@ -18,49 +18,268 @@ static const std::string testName = "LinearRegression";
 
 using namespace mlpack;
 
-// Utility function to set a parameter and mark it as passed, using copy
-// semantics.
-template<typename T>
-void SetInputParam(const std::string& name, const T& value)
-{
-  CLI::GetParam<T>(name) = value;
-  CLI::SetPassed(name);
-}
-
-// Utility function to set a parameter and mark it as passed, using move
-// semantics.
+// Utility function to set a parameter and mark it as passed,
+// using copy semantics for lvalues and move semantics for rvalues
 template<typename T>
 void SetInputParam(const std::string& name, T&& value)
 {
-  CLI::GetParam<T>(name) = std::move(value);
+  CLI::GetParam<typename std::remove_reference<T>::type>(name)
+    = std::forward<T>(value);
   CLI::SetPassed(name);
 }
 
-struct LinearRegressionTestFixture
+struct LRTestFixture
 {
  public:
-  LinearRegressionTestFixture()
+  LRTestFixture()
   {
     // Cache in the options for this program.
     CLI::RestoreSettings(testName);
   }
 
-  ~LinearRegressionTestFixture()
+  ~LRTestFixture()
   {
     // Clear the settings.
     CLI::ClearSettings();
   }
 };
 
-BOOST_FIXTURE_TEST_SUITE(LinearRegressionMainTest, LinearRegressionTestFixture);
+void ResetSettings()
+{
+  CLI::ClearSettings();
+  CLI::RestoreSettings(testName);
+}
+
+BOOST_FIXTURE_TEST_SUITE(LinearRegressionMainTest, LRTestFixture);
+
+/**
+ * Training a model with different regularization parameter
+ * and ensuring that predictions are different
+ */
+BOOST_AUTO_TEST_CASE(LRDifferentLambdas)
+{
+  // required minimal difference between solutions
```

Comment:
This is a really pedantic comment and I can handle it after merge if you don't want to---comments in mlpack should be complete sentences preferably with punctuation and capitalization: https://github.com/mlpack/mlpack/wiki/DesignGuidelines#comments

Reply:
I really wan't to write a good code. Feel free to criticize me. Thank you for detailed review.